sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -205,6 +222,22 @@ trait SimpleFunctionRegistryBase[T] extends FunctionRegistryBase[T] with Logging
     internalRegisterFunction(normalizedName, info, builder)
   }
 
+  override def registerCustomBuiltinFunction(
+      name: FunctionIdentifier,
+      info: ExpressionInfo,
+      builder: FunctionBuilder): Unit = {
+    // We didn't do any check when replacing a function even if it's a builtin one.
+    // So no need to do any extra check here.
+    if (name.database.isEmpty) {
```

Comment:
we actually shouldn't allow function with a database here. Not sure if it's necessary to add a check here because it may break existing use cases.